model/src/main/kotlin/utils/FileArchiver.kt

```diff
@@ -77,6 +87,11 @@ class FileArchiver(
             directory.packZip(zipFile, overwrite = true) { file ->
                 val relativePath = file.relativeTo(directory).invariantSeparatorsPath
 
+                if (!file.isValidTextFile()) {
```

Comment:
I'd just inline the check here as

    tika.detekt(file) != MimeTypes.PLAIN_TEXT